mon-pix/tests/unit/services/modulix-auto-scroll_test.js

```diff
@@ -0,0 +1,68 @@
+import { setupTest } from 'ember-qunit';
+import { module, test } from 'qunit';
+import sinon from 'sinon';
+
+module('Unit | Services | Module | ModulixAutoScroll', function (hooks) {
+  setupTest(hooks);
+
+  module('#setHTMLElementScrollOffsetCssProperty', function () {
+    test('should set --scroll-offset to the given html element', function (assert) {
+      // given
+      const modulixAutoScrollService = this.owner.lookup('service:modulix-auto-scroll');
+      const htmlElement = document.createElement('div');
+
+      // when
+      modulixAutoScrollService.setHTMLElementScrollOffsetCssProperty(htmlElement);
+
+      // then
+      assert.strictEqual(htmlElement.style.cssText, '--scroll-offset: 70px;');
+    });
+  });
+
+  module('#focusAndScroll', function () {
+    test('should call focus on given html element', function (assert) {
+      // given
+      const modulixAutoScrollService = this.owner.lookup('service:modulix-auto-scroll');
+      const htmlElement = document.createElement('div');
+      htmlElement.focus = sinon.stub();
+
+      // when
+      modulixAutoScrollService.focusAndScroll(htmlElement);
+
+      // then
+      sinon.assert.calledWith(htmlElement.focus, { preventScroll: true });
+      assert.ok(true);
+    });
+
+    test('should call scroll to the given html element', function (assert) {
+      // given
+      const scrollOffset = 70;
+      const htmlElementBoundingClientRectTop = 20;
+      const windowScrollY = 5;
+      const calculatedElementY = htmlElementBoundingClientRectTop + windowScrollY;
+      const scrollToTop = calculatedElementY - scrollOffset;
```

Comment:
extraire dans une fonction puis l'appeler directement dans le then

Reply:
À la réflexion, ça ne semble pas utile de refaire ici en détail le calcul qui est fait dans l'implémentation. Dans un test unitaire, on devrait simplement vérifier que pour un ensemble de données passées en argument, la méthode retourne la valeur correct. 

Donc dans dans le "given" on devrait seulement avoir les données "input", soit `scrollOffset` et `htmlElementBoundingClientRectTop` et dans le "then" la donnée "output", soit `scrollToTop`.

Reply:
Comme le scrollOffset n'est pas vraiment une input (on ne passe pas la valeur en argument, c'est une variable d'instance du service), j'ai fait un peu différemment mais je pense que c'est toujours pas ça 😬

Reply:
Je ne comprends pas la nouvelle façon de faire. Du coup on ne teste plus la fonction `focusAndScroll()` du service `modulixAutoScrollService` puisqu'on teste finalement celle implémenté dans le test `executeFocusAndScroll()`? 

Et pourquoi on ne fait plus appel au service avec :       
`const modulixAutoScrollService = this.owner.lookup('service:modulix-auto-scroll');`

Reply:
expliqué en review